test/quantization/pt2e/test_quantize_pt2e.py

```diff
@@ -2562,5 +2562,97 @@ def forward(self, x):
             is_debug_mode=True,
         )
 
+    def test_dynamic_affine_act_per_channel_weights(self):
+        import operator
+
+        from torch.ao.quantization.observer import (
+            MappingType,
+            PerChannelMinMaxObserver,
+            PerToken,
+        )
+        from torch.ao.quantization.pt2e._affine_quantization import (
+            AffineQuantizedMovingAverageMinMaxObserver,
+        )
+
+        class BackendAQuantizer(Quantizer):
+            def annotate(self, model: torch.fx.GraphModule) -> torch.fx.GraphModule:
+                for node in model.graph.nodes:
+                    if (
+                        node.op == "call_function"
+                        and node.target == torch.ops.aten.linear.default
+                    ):
+                        input_act = node.args[0]
+                        assert isinstance(input_act, Node)
+                        weight = node.args[1]
+                        assert isinstance(weight, Node)
+
+                        act_qspec = QuantizationSpec(
+                            dtype=torch.int8,
+                            quant_min=-128,
+                            quant_max=127,
+                            qscheme=None,
+                            is_dynamic=True,
+                            observer_or_fake_quant_ctr=AffineQuantizedMovingAverageMinMaxObserver.with_args(
+                                # TODO: maybe align the arg name here
+                                target_dtype=torch.int8,
```

Comment:
ideally wecan just use `dtype` that's used in L2590

Reply:
Not sure i fully follow. Do you mean that we should move this out to some variable like activation_dtype = torch.int8, and let 

L2590:
dtype = activation_dtype
L2597:
target_dtype = activation_dtype

Reply:
the args from quantization spec are automatically feed into the `observer_or_fake_quant_ctr`: https://github.com/pytorch/pytorch/blob/b0e28f60df6906fa75ff99f6ae64be0bba9fbf33/torch/ao/quantization/fx/prepare.py#L119

Reply:
@jerryzh168 i see, i'll leave consolidating the target_dtype with the quantization spec dtype for later than, as i'm not entirely sure how that should work. I will remove the is_dynamic/quant_min/quant_max calls and change the dtype to activation_dtype.

Reply:
resolving now since its updated. Feel free to reopen this if there is somethign else.